spec/unit/artifactserver_spec.rb

```diff
@@ -56,6 +56,59 @@ module CookbookOmnifetch
       }
       expect(public_repo_location.lock_data).to eq(expected_data)
     end
+
+    context "when asked to install a new cookbook" do
+
+      let(:http_client) { double("Http Client") }
+
+      let(:test_root) { Dir.mktmpdir(nil) }
+
+      let(:storage_path) { File.join(test_root, "storage") }
+
+      let(:cache_path) { File.join(test_root, "cache") }
+
+      let(:path) { fixtures_path.join('cookbooks') }
+
+      let(:cookbook_name) { "example_cookbook" }
+
+      let(:cookbook_version) { "0.5.0" }
+
+      before do
+        allow(CookbookOmnifetch).to receive(:storage_path).and_return(Pathname.new(storage_path))
+        allow(CookbookOmnifetch).to receive(:cache_path).and_return(cache_path)
+        FileUtils.mkdir_p(storage_path)
+      end
+      it "installs the cookbook to the desired install path" do
+        expect(public_repo_location).to receive(:http_client).and_return(http_client)
+        expect(http_client).to receive(:streaming_request) do |arg, &block|
+          Dir.mktmpdir(nil) do |source_dir|
+            gz_file_name = File.join(source_dir, 'input.gz')
+            Zlib::GzipWriter.open(gz_file_name) do |gz|
+              # Minitar writes the full paths provided and doesn't seem to have a way to
+              # remove prefixes.  So we chdir like barbarians.
+              cwd = Dir.getwd
+              Dir.chdir(path)
```

Comment:
You can use `Dir.chdir` with a block. The block will be invoked in the new directory, and when it finishes executing, you'll be returned to where you started (it will also use `ensure` to make sure you're returned where you started even in the case of failure.